### DISCUSSION TITLE
确认一下`HexMeshDataStructure`的local排序

### DISCUSSION THREAD

**Discussion by scaomath** [weihuayi/fealpy#53] [Q&A]
@weihuayi 似乎现在FEALPy里面face的顺序不再像魏老师在ifem里面实现的那样了

这个是ifem
```matlab
elem = [s s+ni s+ni+nij s+nij s+1 s+ni+1 s+ni+nij+1 s+nij+1]; 
totalFace = uint32([elem(:,[1 4 3 2]); elem(:,[1 2 6 5]);...
                               elem(:,[5 6 7 8]); elem(:,[8 7 3 4]);...
                               elem(:,[4 1 5 8]); elem(:,[2 3 7 6])]);
```

这个是现在的FEALPy
https://github.com/weihuayi/fealpy/blob/ea12ef731485079fe736d4fa0c8b80e6c6b40913/fealpy/mesh/StructureHexMesh.py#L74-L77

如果老ifem里面的排序建立这个mesh类，是不是那个类里面的`face2cell`或者`face_to_cell`的函数不能直接用？

**Comment by weihuayi** [comment]
是的，不能直接用。StructureHexMesh.py 中实现 face_to_cell 是按照前面的 localFace 顺序来实现的，但是比较手工化的实现。在非结构的六面体网格中，你给个不同的 localFace，后面的 face_to_cell 仍然可以正确的工作。
